torchgen/gen.py

```diff
@@ -359,11 +360,13 @@ def static_dispatch_extra_headers(backends: List[BackendIndex]) -> List[str]:
     ]
 
 
-# Translates arguments of a native function from DispatcherSignature form to CppSignature form with support for
-# supporting usecases even when there is a memory_format argument along with tensor_option arguments.
-# This usecase is not covered by tools.codegen.api.translate() yet as its application is limited to static dispatch
-def translate_args_dispatcher_to_cpp(
+# Translates arguments of a native function to CppSignature bindings.
+# If `cpp_sig` is provided, use that to retrieve source bindings, otherwise default to DispatcherSignature of `f`.
+# Note that we have a special case for `memory_format` argument and this case is not covered by
+# tools.codegen.api.translate() yet as its application is limited to static dispatch.
+def translate_args(
     f: NativeFunction,
+    cpp_sig: Optional[CppSignature],
```

Comment:
it is better make this not Optional; it is an antipattern to repeat construction of sig on the inside, that means there is not one canonical source of truth